openff/evaluator/_tests/test_backends/test_dask_kubernetes.py

```diff
@@ -0,0 +1,166 @@
+import json
+import pathlib
+
+import pytest
+import yaml
+from openff.units import unit
+from openff.utilities.utilities import get_data_dir_path
+
+from openff.evaluator.backends.backends import PodResources
+from openff.evaluator.backends.dask_kubernetes import (
+    DaskKubernetesBackend,
+    KubernetesEmptyDirVolume,
+    KubernetesSecret,
+)
+
+
+class TestDaskKubernetesBackend:
+    @pytest.fixture
+    def gpu_resources(self):
+        node_affinity = {
+            "nodeAffinity": {
+                "requiredDuringSchedulingIgnoredDuringExecution": {
+                    "nodeSelectorTerms": [
+                        {
+                            "matchExpressions": [
+                                {
+                                    "key": "nvidia.com/cuda.runtime.major",
+                                    "operator": "In",
+                                    "values": ["12"],
+                                },
+                                {
+                                    "key": "nvidia.com/cuda.runtime.minor",
+                                    "operator": "In",
+                                    "values": ["4"],
+                                },
+                            ]
+                        }
+                    ]
+                }
+            }
+        }
+        return PodResources(
+            number_of_threads=1,
+            number_of_gpus=1,
+            affinity_specification=node_affinity,
+            minimum_number_of_workers=1,
+            maximum_number_of_workers=10,
+        )
+
+    @pytest.fixture
+    def cpu_resources(self):
+        return PodResources(
+            number_of_threads=1,
+            number_of_gpus=0,
+            affinity_specification=None,
+            maximum_number_of_workers=20,
+        )
+
+    @pytest.fixture
+    def calculation_backend(self, gpu_resources, cpu_resources):
+        volume = KubernetesEmptyDirVolume(
+            name="evaluator-storage",
+            mount_path="/evaluator-storage",
+        )
+        secret = KubernetesSecret(
+            name="openeye-license",
+            secret_name="oe-license-feb-2024",
+            mount_path="/secrets/oe_license.txt",
+            sub_path="oe_license.txt",
+        )
```

Comment:
This is a very narrow view of a secret, which can be done in [a few ways](https://kubernetes.io/docs/concepts/configuration/secret/) but for an OpenEye license as a file is easiest...